crates/sui-indexer-alt-e2e-tests/src/lib.rs

```diff
@@ -291,13 +263,70 @@ impl FullCluster {
             .await
     }
 
+    pub async fn wait_for_graphql(
+        &self,
+        checkpoint: u64,
+        timeout: Duration,
+    ) -> Result<(), Elapsed> {
+        self.offchain.wait_for_graphql(checkpoint, timeout).await
+    }
+
     /// Triggers cancellation of all downstream services, waits for them to stop, cleans up the
     /// temporary database, and the temporary directory used for ingestion.
     pub async fn stopped(self) {
         self.offchain.stopped().await;
     }
 }
 
+pub struct OffchainClusterConfig {
+    pub indexer_args: IndexerArgs,
+    pub consistent_indexer_args: IndexerArgs,
+    pub client_args: ClientArgs,
+    pub indexer_config: IndexerConfig,
+    pub consistent_config: ConsistentConfig,
+    pub jsonrpc_config: JsonRpcConfig,
+    pub graphql_config: GraphQlConfig,
+    pub registry: prometheus::Registry,
+    pub cancel: CancellationToken,
+    pub ingestion_temp_dir: TempDir,
+    pub stored_genesis: Option<StoredGenesis>,
+}
```

Comment:
- The point of this config type is to give us a convenient way to set up "all default" options, so it should only contain fields that can be defaulted (so, not `client_args`, or `registry` and below).
- There are configs you can set-up that do not require an `ingestion_temp_dir`, but this is always going to create one which is going to cause confusion.